api/v1alpha1/nonadminbackupstoragelocation_types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NonAdminBackupStorageLocationPhase is a simple one high-level summary of the lifecycle of an NonAdminBackupStorageLocation.
+// +kubebuilder:validation:Enum=New;Available;Unavailable;Created;Deleting
+type NonAdminBackupStorageLocationPhase string
+
+// NonAdminBackupStorageLocationPhase constants similar to velerov1.BackupStorageLocationPhase
+const (
+	NaBSLPhaseNew         NonAdminBackupStorageLocationPhase = "New"
+	NaBSLPhaseAvailable   NonAdminBackupStorageLocationPhase = "Available"
+	NaBSLPhaseUnavailable NonAdminBackupStorageLocationPhase = "Unavailable"
+	NaBSLPhaseCreated     NonAdminBackupStorageLocationPhase = "Created"
+	NaBSLPhaseDeleting    NonAdminBackupStorageLocationPhase = "Deleting"
+)
+
+// NonAdminBSLCondition contains addition conditions to the
+// generic ones defined as NonAdminCondition
+// +kubebuilder:validation:Enum=SecretSynced;BSLSynced
```

Comment:
this line should be removed

this type is not referenced in CRD to have the marker

Reply:
Shouldn't matter. We are not exposing this to the user, so it's rather contract with the controller to not set some other values. I will remove that because I don't mind enforcing this one.

Reply:
Last commit removed that. Can we close this as resolved?